### PR TITLE
Jobs JSON endpoint

### PR DESCRIPTION
**Pull request by regiskuckaertz**
Same old same old.

picture 1

### CODE REVIEW

File: commercial/app/controllers/commercial/JobsController.scala [hunk 1]
```diff
@@ -22,14 +26,24 @@ class JobsController(jobsAgent: JobsAgent) extends Controller with implicits.Req
...
-        jsonFormat.result(views.html.jobs.jobs(jobs.take(2), jobSectors, omnitureId, clickMacro))
+        jsonFormat.result(views.html.jobs.jobs(jobs, jobSectors, omnitureId, clickMacro))
       }
     }
   }
+
+  def getJobs = Action { implicit request =>
+      val json = Json.toJson(jobSample(specificIds, segment))
+      Cached(60.seconds){
+        JsonComponent(json)
```
Comment (JonNorman) [inline commercial/app/controllers/commercial/JobsController.scala]: You can now use the thuper thexy [`JsonComponent(jobSample(specificIds, segment))`](https://github.com/guardian/frontend/blob/master/common/app/common/JsonComponent.scala#L17) (merged this morning). It now has a generic constructor so, as long as it has an implicit `Writes` object, you can pass any old thing in.

This saves you converting it to Json yourself, I know, so much effort.

**Review by JonNorman — CHANGES_REQUESTED** [review]
Looks good to me! Apart from the note about the `JsonComponent`, go right ahead.

File: commercial/app/model/commercial/jobs/Job.scala [hunk 2]
```diff
@@ -43,6 +45,28 @@ object Job {
...
+
+  implicit val writesJob: Writes[Job] = (
+    (JsPath \ "id").write[Int] and
+    (JsPath \ "title").write[String] and
+    (JsPath \ "listingUrl").write[String] and
+    (JsPath \ "recruiterLogoUrl").write[String] and
+    (JsPath \ "recruiterName").write[String] and
+    (JsPath \ "locationDescription").write[Option[String]] and
+    (JsPath \ "shortSalaryDescription").write[String]
+  )(unlift(Job.toJsonTuple))
```
Comment (TBonnin) [inline commercial/app/model/commercial/jobs/Job.scala]: You can also create a `new Writes[Job]` and define inside the `writes(job: Job): JsValue` method like done [here](https://github.com/guardian/frontend/blob/b2527874ac8f94f4d61b1bed2d91f0d930b5c590/admin-jobs/app/models/BreakingNews.scala#L30-L42) so you don't have to define this weird intermediary tuple data structure

File: commercial/app/controllers/commercial/JobsController.scala [hunk 3]
```diff
@@ -22,14 +26,24 @@ class JobsController(jobsAgent: JobsAgent) extends Controller with implicits.Req
     JobSector("media", "Media")
   )
 
+  private def jobSample(specificIds: Seq[String], segment: Segment): Seq[Job] =
+    (jobsAgent.specificJobs(specificIds) ++ jobsAgent.jobsTargetedAt(segment)).distinct.take(2)
+
   def renderJobs = Action { implicit request =>
-    (jobsAgent.specificJobs(specificIds) ++ jobsAgent.jobsTargetedAt(segment)).distinct match {
+    jobSample(specificIds, segment) match {
       case Nil => NoCache(jsonFormat.nilResult.result)
```
Comment (TBonnin) [inline commercial/app/controllers/commercial/JobsController.scala]: I know you have not modified it but why are we not caching the empty result (even for a very short period of time)?
Not caching it at all means that if we return no jobs, all requests made to this endpoint will hit our origin as long as we keep returning an empty results.

File: commercial/app/model/commercial/jobs/Job.scala (hunk 2 above)
Comment (regiskuckaertz) [inline commercial/app/model/commercial/jobs/Job.scala]: @TBonnin Thanks! Much cleaner, me likey

File: commercial/app/controllers/commercial/JobsController.scala (hunk 3 above)
Comment (regiskuckaertz) [inline commercial/app/controllers/commercial/JobsController.scala]: Doing that, good call.

**Comment by regiskuckaertz** [comment]
@TBonnin @JonNorman happy with my changes?

**Comment by JonNorman** [comment]
I'm happy with them, but I've made similar changes in my PR https://github.com/guardian/frontend/pull/14466 - might be good to get a second opinion to sense check them ( @TBonnin 😄  )

**Review by JonNorman — APPROVED** [review]

File: commercial/app/controllers/commercial/JobsController.scala [hunk 4]
```diff
@@ -31,7 +31,9 @@ class JobsController(jobsAgent: JobsAgent) extends Controller with implicits.Req
 
   def renderJobs = Action { implicit request =>
     jobSample(specificIds, segment) match {
-      case Nil => NoCache(jsonFormat.nilResult.result)
+      case Nil => Cached(60.seconds) {
+        jsonFormat.nilResult
+      }
```
Comment (TBonnin) [inline commercial/app/controllers/commercial/JobsController.scala]: Don't change it but you can write this in one line. ie: `case Nil => Cached(60)(jsonFormat.nilResult)`

**Review by TBonnin — COMMENTED** [review]
lgtm 👍

**Pull request merged**